src/sgame/sg_bot_ai.cpp

```diff
@@ -795,8 +795,8 @@ AINodeStatus_t BotActionFight( gentity_t *self, AIGenericNode_t *node )
 	}
 
 	// we killed it, yay!
-	if ( !BotEntityIsValidEnemyTarget( self,
-			self->botMind->goal.getTargetedEntity() ) )
+	if ( !mind->goal.targetsValidEntity()
+		|| !BotEntityIsValidEnemyTarget( self, mind->goal.getTargetedEntity() ) )
```

Comment:
BotEntityIsValidEnemyTarget already does a null check

Reply:
It doesn't check if the target is stale though, notably if ent.generation changed. I don't think this has a big chance of breaking because if the generation changed, the target is often dead. But we shouldn't rely on this

Reply:
GentityRef `get()` returns null if the generation changed.

Reply:
Fair enough, I think that's why I didn't check it when I introduced `targetsValidEntity()`. I can remove the commit if you want, but maybe this is better like this.